x-pack/plugins/encrypted_saved_objects/server/saved_objects/encrypted_saved_objects_client_wrapper.ts

```diff
@@ -323,8 +324,8 @@ function getValidId(
     // and not being used to get around the specified ID limitation
     const canSpecifyID = (overwrite && version) || SavedObjectsUtils.isRandomId(id);
     if (!canSpecifyID) {
-      throw new Error(
-        'Predefined IDs are not allowed for saved objects with encrypted attributes, unless the ID has been generated using `SavedObjectsUtils.generateId`.'
+      throw Boom.badRequest(
```

Comment:
Rather than directly coupling to `Boom`, can we instead throw a Saved Objects Error? If you pull this function up into the class, then you'll have access to `errors`, so you can do something like:

```ts
throw this.errors.createBadRequestError('Predefined IDs are not allowed...')
```

Reply:
Good idea, I went ahead and fixed that within https://github.com/elastic/kibana/pull/89814/commits/c79ff9bbfb2a06372912e407526f03093bb397ae.